### PR TITLE
Add IDE pitfall to make sure you have CLI tools installed

### LINKED ISSUE #139: Too easy to miss the tools and CLI tutorials
It's easy to miss the prerequisites in the 3 IDE tutorials.  This is particularly a problem in the [VSCode prerequisites](https://eecs280staff.github.io/tutorials/setup_vscode.html#prerequisites) section because it leads to show-stopper errors later.

It might be nice if there was a yellow pitfall callout to make sure that you did those.

### PR DESCRIPTION
**Pull request by awdeorio**
Add a warning to the IDE tutorials to avoid a common pitfall: failing to install the CLI tools.

Closes #139

### CODE REVIEW

**Comment by amirkamil** [comment]
For Xcode and Visual Studio, should we include links to all three platforms' CLI tools, or just MacOS or Windows, respectively?

File: docs/setup_visualstudio.md [hunk 1]
```diff
@@ -21,6 +21,13 @@ This tutorial recommends command line tools.  To install CLI tools, follow the [
 
 Next, we recommend our [Command line interface (CLI)](cli.html) tutorial.
 
+<div class="primer-spec-callout warning" markdown="1">
+**Pitfall:** Make sure you have installed CLI tools for your OS before continuing.
```
Comment (ohjuny) [inline docs/setup_visualstudio.md:25]:
```suggestion
**Pitfall:** Make sure you have installed [CLI tools for Windows](setup_wsl.html) before continuing.
```

File: docs/setup_visualstudio.md [hunk 2]
```diff
@@ -21,6 +21,13 @@ This tutorial recommends command line tools.  To install CLI tools, follow the [
...
+
+| [macOS](setup_macos.html)| [Windows](setup_wsl.html) | [Linux](setup_wsl.html#install-cli-tools)
```
Comment (ohjuny) [inline docs/setup_visualstudio.md:27]:
```suggestion
```

File: docs/setup_xcode.md [hunk 3]
```diff
@@ -21,6 +21,13 @@ This tutorial recommends command line tools.  To install CLI tools, follow the [
 
 Next, we recommend our [Command line interface (CLI)](cli.html) tutorial.
 
+<div class="primer-spec-callout warning" markdown="1">
+**Pitfall:** Make sure you have installed CLI tools for your OS before continuing.
```
Comment (ohjuny) [inline docs/setup_xcode.md:25]:
```suggestion
**Pitfall:** Make sure you have installed [CLI tools for macOS](setup_macos.html) before continuing.
```

File: docs/setup_xcode.md [hunk 4]
```diff
@@ -21,6 +21,13 @@ This tutorial recommends command line tools.  To install CLI tools, follow the [
...
+
+| [macOS](setup_macos.html)| [Windows](setup_wsl.html) | [Linux](setup_wsl.html#install-cli-tools)
```
Comment (ohjuny) [inline docs/setup_xcode.md:27]:
```suggestion
```

**Review by ohjuny — APPROVED** [review]
I think Professor Kamil's suggestion will help avoid some potential minor confusion.

File: docs/setup_vscode.md [hunk 5]
```diff
@@ -33,6 +33,14 @@ GNU gdb (GDB)
 
 Next, follow our [Command line interface (CLI)](cli.html) tutorial.
 
+<div class="primer-spec-callout warning" markdown="1">
+**Pitfall:** Make sure you have installed CLI tools for your OS before continuing.
+
+| [macOS](setup_macos.html)| [Windows](setup_wsl.html) | [Linux](setup_wsl.html#install-cli-tools)
```
Comment (ohjuny) [inline docs/setup_vscode.md:39]:
```suggestion
| [macOS](setup_macos.html#install-cli-tools)| [Windows](setup_wsl.html#install-cli-tools) | [Linux](setup_wsl.html#install-cli-tools)
```

**Review by ohjuny — COMMENTED** [review]
Updated links to go straight to #install-cli-tools

**Review by awdeorio — COMMENTED** [review]
Thanks for the code suggestions @ohjuny !

**Pull request merged**